Navigation/NavigationState.js

```diff
@@ -19,6 +20,11 @@ export type NavigationRoute = string | {
   routes: ?Array<NavigationRoute>;
 };
 
+export const NavigationStateShape = React.PropTypes.shape({
+  routes: React.PropTypes.array.isRequired,
```

Comment:
I suppose thats true, but I would not want to prevent any legitimate but unthought of use cases either. cc @ericvicenti